### PR TITLE
Device handlers cleanup

### PR DESCRIPTION
**Pull request by drwill-ms**
InnerHandler means what? It is the next handler in the chain, right? I've renamed it to make more sense.

Removed the timeout methods.

### CODE REVIEW

File: iothub/device/src/InternalClient.cs [hunk 1]
```diff
@@ -465,7 +465,7 @@ public async Task CompleteMessageAsync(string lockToken, CancellationToken cance
 
             try
             {
-                await InnerHandler.CompleteAsync(lockToken, cancellationToken).ConfigureAwait(false);
+                await InnerHandler.CompleteMessageAsync(lockToken, cancellationToken).ConfigureAwait(false);
```
Comment (drwill-ms) [inline iothub/device/src/InternalClient.cs:468]: I renamed this in the clients, but missed the handlers. Now C2D methods make more sense as being related to messaging.

File: iothub/device/src/Pipeline/DefaultDelegatingHandler.cs [hunk 2]
```diff
@@ -10,59 +10,51 @@ namespace Microsoft.Azure.Devices.Client.Transport
 {
     internal abstract class DefaultDelegatingHandler : IDelegatingHandler
     {
-        private volatile IDelegatingHandler _innerHandler;
+        private volatile IDelegatingHandler _nextHandler;
```
Comment (andyk-ms) [inline iothub/device/src/Pipeline/DefaultDelegatingHandler.cs:13]: It is private anyway, but wonder if 'next' does communicate 'delegating'.
Why don't we call it 'delegate' directly (delegating to this)?

**Review by andyk-ms — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/service/src/IotHubClientWebSocket.cs [hunk 3]
```diff
@@ -241,7 +241,7 @@ public async Task<int> ReceiveAsync(byte[] buffer, int offset, TimeSpan timeout)
                         State = WebSocketState.Closed;
                         WebSocketStream.Close();
                         TcpClient.Close();
-                        return 0;  // TODO: throw exception?
+                        return 0; // TODO: throw exception?
```
Comment (abhipsaMisra) [inline iothub/service/src/IotHubClientWebSocket.cs:244]: Another task for us to address

File: iothub/device/src/Pipeline/DefaultDelegatingHandler.cs (hunk 2 above)
Comment (drwill-ms) [reply iothub/device/src/Pipeline/DefaultDelegatingHandler.cs:13]: What is the delegating part? I've never understood that either.

It is a chain of handlers for an operation.

File: iothub/service/src/IotHubClientWebSocket.cs (hunk 3 above)
Comment (drwill-ms) [reply iothub/service/src/IotHubClientWebSocket.cs:244]: Seems like it.

**Pull request merged**